packages/gatsby/cache-dir/loader.js

```diff
@@ -2,6 +2,8 @@ import prefetchHelper from "./prefetch"
 import emitter from "./emitter"
 import { setMatchPaths, findPath, findMatchPath } from "./find-path"
 
+import { createFromReadableStream } from "react-server-dom-webpack"
```

Comment:
will this be bundled for gatsby@4 line potentially making bundles we produce heavier?

Reply:
I think it gets tree-shaken but i'll check to be sure

Reply:
Unfortunately it's not:

![image](https://user-images.githubusercontent.com/419821/191264274-b94c03b9-bded-45cc-b385-7e84082ec711.png)

and 2 options I see is
 1. either make changes needed for tree-shaking to actually work and remove that from bundle
 2. and/or do something like `null-loader` or shim for `react-server-dom-webpack` when partial hydration is not enabled

I feel a bit uneasy relying just on terser to remove uneeded deps tbh because that might be very easy to regress without much notice

Reply:
@pieh @tyhopp @LekoArts this can be Tomorrow making moves topic maybe?

Reply:
@pieh fixed in [`edc770b` (#36485)](https://github.com/gatsbyjs/gatsby/pull/36485/commits/edc770b9f8a4a0eeb648f9610fb8dfcea0d6a058)